### DISCUSSION TITLE
How to start or reload a new playlist immediately ?

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#2407] [General]
Hi, I'm using the watch mode on the playlist.

When I make changes to the playlist, it's reloaded automatically but not start playing the new content, only after one/two musics from the old playlist. How to start immediately the new playlist content from the beginning?

Another case is if y change the playlist with the comando xxx.uri the same occurs, the new playlist start after one/two musics from the old playlist, how to start immediately the new playlist content from the beginning after running the telnet command xxx.uri ?

My code:

`playlist = playlist(reload_mode="watch", loop=true, mode="normal", "/home/test.pls")`

Telnet command that I'm using:

`test.pls.uri /home/new.pls`

**Comment by toots** [comment]
You need to empty the queue if you want to make sure that it starts playing new content right away. If you're using telnet, you should be able to register your own command:
```ruby
server.register(namespace=playlist.id(), description="Flush the playlist's internal queue",
  "fush_queue", fun (_) -> begin
    playlist.set_queue([])
    "Done"
    end)
```
Then:
```
test.pls.uri /home/new.pls
test.pls.flush_queue
```

**Comment by caf2022** [comment]
Hi, when I try to add the server.register to my LS conf I get this error when start LS:

```
    playlist.set_queue([])

Error 5: this value has no method set_queue
```

LS version 2.0.4

**Reply by caf2022** [reply]
I have changed the position of that code to right after the playlist() and now is working. Thanks a lot for the help.
